src/hotspot/cpu/aarch64/icBuffer_aarch64.cpp

```diff
@@ -52,7 +52,7 @@ void InlineCacheBuffer::assemble_ic_buffer_code(address code_begin, void* cached
   address start = __ pc();
   Label l;
   __ ldr(rscratch2, l);
-  __ far_jump(ExternalAddress(entry_point));
+  __ far_jump(ExternalAddress(entry_point), NULL, rscratch1, true);
```

Comment:
This complicates `assemble_ic_buffer_code`. You need to know `far_jump` implementation, especially the generation of  NOPs. I understand why we need those NOPs.
Do we have calls of non-nmethod code here?

Reply:
Yes, there are entry points from both non_method and method segments.

Reply:
I suggest the following code:
```
int inst_count = __ far_jump(ExternalAddress(entry_point));
// IC stub code size is not expected to vary depending on target address.
// We use NOPs to make the size equal to ic_stub_code_size.
for (int i = 3 + inst_count; i < ic_stub_code_size(); ++i) {
  __ nop();
}
```